class-wc-facebookcommerce.php

```diff
@@ -383,8 +384,9 @@ public function add_wordpress_integration() {
 		 * @param string $log_id optional log id to segment the files by, defaults to plugin id
 		 */
 		public function log( $message, $log_id = null ) {
-			// bail if logging isn't enabled
-			if ( ! $this->get_integration() || ! $this->get_integration()->is_debug_mode_enabled() ) {
+			// Bail if site is connected and user has disabled logging.
+			// If site is disconnected, force-enable logging so merchant can diagnose connection issues.
+			if ( ( ! $this->get_integration() || ! $this->get_integration()->is_debug_mode_enabled() ) && $this->get_connection_handler()->is_connected() ) {
```

Comment:
Do we have other places that would start logging stuff when there is no connection?

Reply:
Did a quick search and didn't anything rang a bell, I wonder why you ask? Will merge this PR and we can expand on this if necessary on another PR 💯